SampleScripts/PowerManagement-ManagementDomain.ps1

```
@@ -1032,7 +1205,145 @@ if ($PsBoundParameters.ContainsKey("startup")) {
+                                if ( (Test-VsanHealth -cluster $clustername -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass) -ne 0) {
+                                    Write-PowerManagementLogMessage -Type ERROR -Message "vSAN cluster health is bad. Check your environment and run the script again."
+                                    Exit
+                                }
+                                # Check vSAN Status
+                                if ( (Test-VsanObjectResync -cluster $clustername -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass) -ne 0) {
+                                    Write-PowerManagementLogMessage -Type ERROR -Message "vSAN object resynchronization is in progress. Check your environment and run the script again."
+                                    Exit
+                                }
+
+                                #Start workflow for VCF prior version 4.5
+                                if ([float]$vcfVersion -lt [float]4.5) {
+                                    # Start vSphere HA
+                                    if (!$(Set-VsphereHA -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass -cluster $clustername -enableHA)) {
+                                        Write-PowerManagementLogMessage -Type ERROR -Message "Could not enable vSphere High Availability for cluster '$cluster'."
+                                    }
+
+                                    # Restore the DRS Automation Level to the mode backed up for Management Domain Cluster during shutdown
+                                    if ([string]::IsNullOrEmpty($DrsAutomationLevel)) {
+                                        Write-PowerManagementLogMessage -Type ERROR -Message "The DrsAutomationLevel value in the JSON file is empty. Exiting!"
```

```suggestion
                                        Write-PowerManagementLogMessage -Type ERROR -Message "The DrsAutomationLevel value in the JSON file is empty. Exiting..."
```

Should this be checked earlier to fail faster?